src/routes/tools/extreme-precipitation/_store.js

```diff
@@ -0,0 +1,101 @@
+import { writable, derived } from "svelte/store";
+import climvars from "~/helpers/climate-variables";
+import { makeCustomWritableStore } from "../_common/stores";
+import {
+  CLIMATE_INDICATORS,
+  DEFAULT_CLIMATE_VARIABLE,
+  DEFAULT_CLIMATE_INDICATOR,
+  DEFAULT_DURATION,
+  DEFAULT_RETURN_PERIOD,
+  DEFAULT_THRESHOLD_TYPE,
+} from "./_constants";
+
+import {
+  calcHeatwaveCount,
+  calcMaxDuration,
+  calcDaysCount,
+  groupDataByYear,
+} from "./_data";
+
+export const climvarStore = makeCustomWritableStore(DEFAULT_CLIMATE_VARIABLE, {
+  name: "climvarStore",
+  getters: [
+    {
+      name: "climvar",
+      getter: ($s) => climvars.find((d) => d.id === $s),
+    },
+  ],
+});
+
+export const indicatorList = CLIMATE_INDICATORS;
+
+export const indicatorStore = (() => {
+  const store = writable(DEFAULT_CLIMATE_INDICATOR);
+  const { set, subscribe } = store;
+  return {
+    set,
+    subscribe,
+    get indicator() {
```

Comment:
I'm curious if using `makeCustomWritableStore` does not work for this use case?

Reply:
Forgot about that one. Fixed in https://github.com/berkeley-gif/caladapt-website-2021/pull/163/commits/fa01f01981bcfa571605ac8778a0911054ea9b3f